### PR TITLE
Bump pyBigWig to 0.3.12

### PR DESCRIPTION
**Pull request by dpryan79**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/pybigwig/meta.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ requirements:
     - curl
     - libcurl
     - zlib
-    - numpy
+    - {{ pin_compatible('numpy') }}
```
Comment (bgruening) [inline recipes/pybigwig/meta.yaml]: I don't think this is correct. Any reason why you have changed this? 
xref: https://conda-forge.org/docs/meta.html#building-against-numpy

Comment (dpryan79) [reply recipes/pybigwig/meta.yaml]: Click over to the circleci error from the previous commit. For whatever reason it appears to have been building against one version (1.9.3 for some reason) and then testing/running against another (1.15.0). That won't work in C unless things are ABI compatible. This change seemed odd, but it got everything to the same version.

Comment (dpryan79) [reply recipes/pybigwig/meta.yaml]: https://circleci.com/gh/bioconda/bioconda-recipes/23223?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by bgruening** [comment]
Mh ... https://github.com/conda-forge/conda-forge-pinning-feedstock/blame/master/recipe/conda_build_config.yaml#L424 the idea is really to compile against an old version that is ABI compatible and install the latest version.

**Comment by dpryan79** [comment]
It's probably a bug in the header of that old version of numpy (I recall looking for the cause locally). `PyInt_Check()` is only defined in python2, since the size of integers changed in python 3. It appears that this bug was simply fixed in the more recent version that I compiled against.

**Comment by dpryan79** [comment]
Here's the relevant bug in numpy: https://github.com/numpy/numpy/blob/maintenance/1.9.x/numpy/core/include/numpy/ndarrayobject.h#L68

That was fixed in numpy 1.10 (as best I can tell), which appears to have been the first to really support python 3 properly in the headers.

**Comment by dpryan79** [comment]
I should note that I never ran into this before because I didn't use the `PyArray_IsIntegerScalar()` function before, since I didn't have any users using pandas before.

**Pull request merged**